lib/lineage.js

```diff
@@ -91,7 +112,13 @@ const fetchHydratedDoc = id => {
           fillContactsInDocs(lineage, allContacts);
           const doc = lineage.shift();
           buildHydratedDoc(doc, lineage);
-          return doc;
+
+          // Also attach the patient if we should
+          if (doc.type === 'data_record' && doc.fields && doc.fields.patient_id) {
```

Comment:
Might you also want to do this for registrations where the `patient_id` is generated not submitted and therefore not under the `fields` property?